### PR TITLE
[SD-2564] [SD-3493] Import document to legal archive after publish actions.

### PR DESCRIPTION
**Pull request by mdhaman**
*Change includes:*
- Moving an item into legal archive on publish actions (except scheduling an item). 
- For schedule publish, the moving into legal archive will be handled by the async publishing PR (@mugurrus).
- Do we need a `Date` parameter for ImportLegalPublishQueueCommand??

### CODE REVIEW

File: server/apps/legal_archive/commands.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+        legal_archive_doc.pop('lock_session', None)
+        legal_archive_doc.pop('lock_time', None)
+
+        logger.info('Removed irrelevant properties from the article ' + log_msg)
+
+        # Step 1
+        article_in_legal_archive = legal_archive_service.find_one(req=None, _id=legal_archive_doc[config.ID_FIELD])
+
+        # if there a version in that has version greater then current doc then return
+        if article_in_legal_archive and article_in_legal_archive[config.VERSION] >= legal_archive_doc[config.VERSION]:
```
Comment (sivakuna-aap) [inline server/apps/legal_archive/commands.py]: Ideally the version in legal archive will never be greater than the doc in archive.

File: server/apps/legal_archive/commands.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+
+    def upsert_into_legal_archive(self, doc):
+        """
+        Once publish actions are performed on the article do the below:
+            1.  Check if the legal_archive is up-to-date or not.
+            2.  De-normalize the expired article
+            3.  Fetch version history of article so that version_history_doc[config.VERSION] <= doc[config.VERSION]
+                and De-normalize.
+            4.  Upserting Legal Archive.
+            5.  Inserting Legal Archive Versions.
```
Comment (sivakuna-aap) [inline server/apps/legal_archive/commands.py]: Even the versions should be denormalized

**Comment by ioanpocol** [comment]
File "/home/travis/build/superdesk/superdesk/server/apps/publish/published_item.py", line 19, in <module>
    from apps.legal_archive import LEGAL_ARCHIVE_NAME, LEGAL_ARCHIVE_VERSIONS_NAME, LEGAL_PUBLISH_QUEUE_NAME

ImportError: cannot import name 'LEGAL_ARCHIVE_NAME'

**Comment by akintolga** [comment]
:+1:  Imo a date parameter would be handy target a subset of publish queue items but not urgent.

**Comment by ioanpocol** [comment]
:+1:

**Pull request merged**